sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobItem.java

```diff
@@ -15,6 +17,30 @@
 @JacksonXmlRootElement(localName = "Blob")
 @Fluent
 public final class BlobItem {
+
+    /**
+     * Initializes a new BlobItem.
+     */
+    public BlobItem() {
+
+    }
+
+    /**
+     * Initializes a new blob item.
+     *
+     * @param blobItemInternal The internal structure from which to pull state.
+     */
+    public BlobItem(BlobItemInternal blobItemInternal) {
```

Comment:
Is this ok to do since BlobItemInternal is "private" and BlobItem is public. Would it be better to like extract the pieces in a private method wherever you use this constructor and make this constructor like take in all the properties?

Reply:
Though this is technically cleaner

Reply:
Hmm. Interesting. I guess that is what we do with BlobProperties. Though I don't remember the reason why. There must've been a reason other than we needed to construct it from different sources (download, getProperties, etc.) or we would've just had multiple constructors.